platformio.ini

```diff
@@ -22,24 +31,37 @@ lib_deps =
     https://github.com/bblanchon/ArduinoJson#3df4efd
     https://github.com/Makuna/NeoPixelBus#9619fef
 
-[env:nodemcuv2]
-platform = espressif8266@2.0.0 ; Corresponds to https://github.com/platformio/platform-espressif8266/releases/tag/v2.0.0 = https://github.com/esp8266/Arduino/releases/tag/2.5.0
-board = nodemcuv2
-framework = arduino
+
+# base environments
+# can be extended by board-specific environments
+[env:esp8266_base]
+; corresponds to https://github.com/platformio/platform-espressif8266/releases/tag/v2.0.0
+; see https://github.com/esp8266/Arduino/releases/tag/2.5.0
+platform = espressif8266@2.0.0
+framework = ${common.framework}
+lib_deps = ${common.lib_deps}
+
+[env:esp32_base]
+; corresponds to https://github.com/platformio/platform-espressif32/releases/tag/v1.8.0
+; see https://github.com/espressif/arduino-esp32/releases/tag/1.0.2
+platform = espressif32
+framework = ${common.framework}
 lib_deps =
     ${common.lib_deps}
+    https://github.com/bbx10/Hash_tng
+
+
+[env:nodemcuv2]
+board = nodemcuv2
+extends = env:esp8266_base
 
 [env:d1_mini]
-platform = espressif8266@2.0.0 ; Corresponds to https://github.com/platformio/platform-espressif8266/releases/tag/v2.0.0 = https://github.com/esp8266/Arduino/releases/tag/2.5.0
 board = d1_mini
-framework = arduino
-lib_deps =
-    ${common.lib_deps}
+extends = env:esp8266_base
 
+# esp32dev works for the majority of ESP32 based dev boards
+# there are more specific board configurations available, feel free to send PRs adding new ones
 [env:esp32dev]
-platform = espressif32@1.8.0 ; Corresponds to https://github.com/platformio/platform-espressif32/releases/tag/v1.8.0 = https://github.com/espressif/arduino-esp32/releases/tag/1.0.2
```

Comment:
Why was this pinning removed, by mistake?

Reply:
Check the `extends` line. It points you to the environment `env:esp32_base`, where this line is contained. What's missing is the `@1.8.0` there. I'll update the PR.

Reply:
Actually, the `@1.8.0` was wrong anyway. It should've been `1.0.2`

Reply:
Strike that, `1.8.0` is the right one.